app/src/main/java/com/github/giommok/softwaredevproject/FirebaseAccount.java

```diff
@@ -93,97 +115,160 @@ public String getUsername() {
+            if(!discoveredPeaks.contains(peak)) discoveredPeaks.add(peak);
+        }
+    }
+
+    /**
+     * Get the username from the database
+     * Assign the username attribute with its value
+     * @param usernameFromHashMap username stored in the database
+     */
+    private static void syncGetUsernameFromProfile(String usernameFromHashMap){
+       username = usernameFromHashMap;
+    }
+
+    /**
+     * Get the user score from the database
+     * Assign the score attribute with its value
+     * @param scoreFromHashMap user score stored in the database
+     */
+    private static void syncGetUserScoreFromProfile(long scoreFromHashMap){
+        score = scoreFromHashMap;
+    }
+
+    /**
+     * Get the list of discovered country highPoints from the datasnapshot from database.
+     * The list is automatically appended to the discoveredCountryHighPoint HashSet
+     * @param countryHighPointFromHashMap snapshot from the database userchild
+     */
+    private static void syncGetCountryHighPointsFromProfile(DataSnapshot countryHighPointFromHashMap){
+
+        HashMap<String, HashMap<String, String>> entries = (HashMap<String, HashMap<String, String>>) countryHighPointFromHashMap.getValue();
+
+        //Iterate over the data snapshot to re-create objects
+        for (Map.Entry<String, HashMap<String, String>> entry : entries.entrySet()) {
+            Object value = entry.getValue();
+            CountryHighPoint countryHighPoint = new CountryHighPoint(((HashMap<String, String>) value).get(Database.CHILD_ATTRIBUTE_COUNTRY_NAME),
+                    ((HashMap<String, String>) value).get(Database.CHILD_COUNTRY_HIGH_POINT),
+                    ((HashMap<String, Long>) value).get(Database.CHILD_ATTRIBUTE_HIGH_POINT_HEIGHT));
+            String countryName = ((HashMap<String, String>) value).get(Database.CHILD_ATTRIBUTE_COUNTRY_NAME);
+            //Check if the country high point is already in the list to avoid duplicate
+            if(!discoveredCountryHighPoint.containsKey(countryName))
+                discoveredCountryHighPoint.put(((HashMap<String, String>) value).get(Database.CHILD_ATTRIBUTE_COUNTRY_NAME), countryHighPoint);
+        }
+    }
+
+    /**
+     * Get the list of discovered peak heights (ranges)
+     * The list is automatically appended to the discoveredHeights hashSet
+     * @param discoveredHeightsFromHashMap snapshot from the database userchild
+     */
+    private static void syncGetDiscoveredHeight(DataSnapshot discoveredHeightsFromHashMap){
+        HashMap<String, HashMap<String, String>> entries = (HashMap<String, HashMap<String, String>>) discoveredHeightsFromHashMap.getValue();
+        for (Map.Entry<String, HashMap<String, String>> entry : entries.entrySet()){
+            Object value = entry.getValue();
+            long retrievedVal = ((ArrayList<Long>) value).get(0);
+            if(!discoveredPeakHeights.contains(retrievedVal)) discoveredPeakHeights.add((int)retrievedVal);
```

Comment:
Same here for the .add()
```suggestion
             discoveredPeakHeights.add((int)retrievedVal);
```